lib/pf/ConfigStore/Authentication.pm

```diff
@@ -15,23 +15,174 @@ pf::ConfigStore::Authentication
 use Moo;
 use namespace::autoclean;
 use pf::authentication;
+use pf::file_paths;
 use HTTP::Status qw(:constants is_error is_success);
+use pf::authentication;
+use pf::log;
+use pfconfig::cached_hash;
+use List::Util qw(first);
+use List::MoreUtils qw(none any);
+
+
+our $cached_authentication_config;
+our @auth_sources;
+our %auth_lookup;
 
 extends 'pf::ConfigStore';
 
+readAuthenticationConfigFile();
+setModuleSources();
+
 =head1 METHODS
 
 =head2 _buildCachedConfig
 
 =cut
 
-sub _buildCachedConfig { $pf::authentication::cached_authentication_config };
+sub _buildCachedConfig { $cached_authentication_config };
+
+sub configFile { $authentication_config_file }
+
+sub pfconfigNamespace {'config::Authentication'}
 
 before rewriteConfig => sub {
     my ($self) = @_;
     $self->cachedConfig->ReorderByGroup();
 };
 
+sub readAuthenticationConfigFile {
+    unless ($cached_authentication_config) {
+        $cached_authentication_config = pf::config::cached->new (
+            -file => $authentication_config_file,
+            -oncachereload => [
+              on_cache_authentication_reload => sub {setModuleSources()}
+        ] );
+    } else {
+        $cached_authentication_config->ReadConfig();
+    }
+}
+
+sub setModuleSources {
+    @auth_sources = sub { return @pf::authentication::authentication_sources }->();
+    %auth_lookup = map { $_->id => $_ } grep { defined $_ } @auth_sources;
+}
+
+sub getSource {
+    my ($id) = @_;;
+    return unless defined $id && exists $auth_lookup{$id};
+
+    return $auth_lookup{$id};
+}
+
+=item deleteSource
+
+Delete an authentication source along its rules. Returns the number of source(s)
+deleted.
+
+=cut
+
+sub deleteSource {
+    my $id = shift;
+    my $logger = get_logger;
+
+    my %Profiles_Config;
+    tie %Profiles_Config, 'pfconfig::cached_hash', 'config::Profiles';
+
+    my $result = 0;
+    if (none { any {$_ eq $id} @{$_->{sources}} } values %Profiles_Config) {
```

Comment:
Could the original author have been any less clear?
Please, please rework that...